src/base_alloc/base_alloc_linear.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+#include <stdint.h>
+
+#include "base_alloc_internal.h"
+#include "base_alloc_linear.h"
+#include "utils_common.h"
+#include "utils_concurrency.h"
+
+// minimum size of a single pool of the linear base allocator
+#define MINIMUM_LINEAR_POOL_SIZE (ba_os_get_page_size())
+
+// alignment of the linear base allocator
+#define MEMORY_ALIGNMENT (8)
+
+typedef struct umf_ba_next_linear_pool_t umf_ba_next_linear_pool_t;
```

Comment:
Generally, I saw many years ago '_t' style and there was something like:
`struct umf_ba_next_linear_pool` or `umf_ba_next_linear_pool_t`
So, I think there are two possibilities:
```c
struct umf_ba_next_linear_pool {
};
typedef umf_ba_next_linear_pool umf_ba_next_linear_pool_t;
```
or in one-line:
```c
typedef struct {} umf_ba_next_linear_pool_t;
```

Reply:
Done